resolwe_bio/processes/import_data/sra_file.py

```diff
@@ -0,0 +1,403 @@
+
+            fastqc_url.append({
+                "file": str(report_dir / 'fastqc_report.html'),
+                "refs": [str(report_dir)],
+            })
+
+            reads_name = report_dir.name.replace('_fastqc', '')
+            encoding_file = report_dir / 'fastqc_data.txt'
+            encoding = Cmd['parse_encoding_type.py'](str(encoding_file))
+
+            if encoding == 'Illumina 1.5' or encoding == 'Illumina 1.3':
+                print("Recoding input reads from Phred64 encoding to Phred33 encoding.")
+                os.rename(f'{reads_name}.fastq.gz', 'input_reads.fastq.gz')
+                cmd = Cmd[
+                    'TrimmomaticSE',
+                    '-phred64',
+                    'input_reads.fastq.gz',
+                    'reformated.fastq.gz',
+                    'TOPHRED33',
+                ]()
+                os.rename('reformated.fastq.gz', f'{reads_name}.fastq.gz')
+
+            elif encoding != "Sanger / Illumina 1.9\n":
+                self.error(
+                    "Only Sanger / Illumina 1.9 / llumina 1.5 / Illumina 1.3 encoding is "
+                    "supported."
+                )
+
+        fastqc = glob.glob('*_fastqc.zip')
+
+        outputs.fastqc_url = fastqc_url
+        outputs.fastqc_archive = fastqc
+
+
+class ImportSraPaired(Process):
+    """Import paired-end reads from SRA.
+
+    Import paired-end reads from Sequence Read Archive (SRA) via an SRA accession number.
+    SRA stores raw sequencing data and alignment information from high-throughput
+    sequencing platforms.
+
+    """
+    slug = 'import-sra-paired'
+    name = 'SRA data (paired-end)'
+    process_type = 'data:reads:fastq:paired'
+    version = '1.0.0'
+    category = 'Import'
+    scheduling_class = SchedulingClass.BATCH
+    persistence = Persistence.RAW
+    entity = {
+        'type': 'sample',
+        'descriptor_schema': 'sample',
+    }
+    requirements = {
+        'expression-engine': 'jinja',
+        'executor': {
+            'docker': {
+                'image': 'resolwebio/common:1.3.1'
+            }
+        },
+        'resources': {
+            'cores': 1,
+            'memory': 1024,
+            'network': True,
+        },
+    }
+    data_name = '{{ sra_accession|first }}'
+
+    class Input:
+        """Input fields to process ImportSraPaired."""
+        sra_accession = ListField(StringField(), label='SRA accession(s)')
+        show_advanced = BooleanField(label='Show advanced options', default=False)
+
+        class Advanced:
+            """Advanced options."""
+            min_spot_id = IntegerField(label='Minimum spot ID', required=False)
+            max_spot_id = IntegerField(label='Maximum spot ID', required=False)
+            min_read_len = IntegerField(label='Minimum read length', required=False)
+            clip = BooleanField(label='Clip adapter sequences', default=False)
+            aligned = BooleanField(label='Dump only aligned sequences', default=False)
```

Comment:
Are these two options (`aligned`, `unaligned`) mutually exclusive? If yes, can we make them like that here?

Reply:
@mzganec